sdk/src/client/builder.rs

```diff
@@ -351,6 +351,33 @@ pub struct NetworkInfo {
     pub latest_milestone_timestamp: Option<u32>,
 }
 
+impl NetworkInfo {
+    pub fn with_protocol_parameters(mut self, protocol_parameters: impl Into<ProtocolParameters>) -> Self {
+        self.protocol_parameters = protocol_parameters.into();
+        self
+    }
+
+    pub fn with_local_pow(mut self, local_pow: bool) -> Self {
```

Comment:
There are instances in the code base where we set fields directly, should they start using these setters?

Reply:
Doesn't really matter

Reply:
Same answer, not sure why we're adding setters if we keep the fields public, they are basically useless.